pandas/tests/indexes/test_base.py

```diff
@@ -2301,9 +2301,16 @@ def test_ensure_index_from_sequences(self, data, names, expected):
         tm.assert_index_equal(result, expected)
 
 
-@pytest.mark.parametrize('opname', ['eq', 'ne', 'le', 'lt', 'ge', 'gt'])
+@pytest.mark.parametrize('opname', ['eq', 'ne', 'le', 'lt', 'ge', 'gt',
+                                    'add', 'radd', 'sub', 'rsub',
+                                    'mul', 'rmul', 'truediv', 'rtruediv',
+                                    'floordiv', 'rfloordiv',
+                                    'pow', 'rpow', 'mod', 'divmod'])
 def test_generated_op_names(opname, indices):
     index = indices
+    if type(index) is pd.Index and opname == 'rsub':
```

Comment:
use isinstance, or is this a very specific check if some is NOT a Index. maybe better to use ABCIndex and/or ABCIndexClass

Reply:
This is specific to pd.Index, not subclasses.

Reply:
pls change this